packages/web-react/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useRef, MutableRefObject, Children, cloneElement, forwardRef } from 'react';
+import { DialogProps, DOMRef } from '../../types';
+import { useDialog } from './useDialog';
+import { useCancelEvent, useClickOutside } from '../../hooks';
+
+const Dialog = (props: DialogProps, ref: DOMRef): JSX.Element => {
+  const { children, isOpen, onClose, ...restProps } = props;
+  // @TODO: fix types later
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const dialogElementRef: MutableRefObject<HTMLDialogElement | null> = useRef(ref);
```

Comment:
@tomassychra Maybe you will also have the suggestion for typing forwarded ref which is used as the initial value.

Reply:
Maybe try this
`const dialogElementRef: MutableRefObject<ForwardedRef<HTMLDialogElement | null>> = useRef(ref);`
and then set `dialogElementRef as MutableRefObject<HTMLDialogElement | null>`

I know it's probably not very nice, but I think it's definitely better than // @ts-ignore

Reply:
Ok, I am not satisfied with using `as` but this time and for this purpose, it is better than ignoring. Thanks for the suggestion.